sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor{TPartition}.cs

```diff
@@ -0,0 +1,433 @@
+using Azure.Messaging.EventHubs.Processor;
+
+namespace Azure.Messaging.EventHubs.Primitives
+{
+    /// <summary>
+    ///   Provides a base for creating a custom processor for consuming events across all partitions of a given Event Hub
+    ///   within the scope of a specific consumer group.  The processor is capable of collaborating with other instances for
+    ///   the same Event Hub and consumer group pairing to share work by using a common storage platform to communicate.  Fault
+    ///   tolerance is also built-in, allowing the processor to be resilient in the face of errors.
+    /// </summary>
+    ///
+    /// <typeparam name="TPartition">The context of the partition for which an operation is being performed.</typeparam>
+    ///
+    public abstract class EventProcessor<TPartition> where TPartition : EventProcessorPartition, new()
+    {
+        /// <summary>Indicates whether or not this event processor is currently running.  Used only for mocking purposes.</summary>
+        private bool? _isRunningOverride;
+
+        /// <summary>
+        ///   The fully qualified Event Hubs namespace that the processor is associated with.  This is likely
+        ///   to be similar to <c>{yournamespace}.servicebus.windows.net</c>.
+        /// </summary>
+        ///
+        public string FullyQualifiedNamespace { get; }
+
+        /// <summary>
+        ///   The name of the Event Hub that the processor is connected to, specific to the
+        ///   Event Hubs namespace that contains it.
+        /// </summary>
+        ///
+        public string EventHubName { get; }
+
+        /// <summary>
+        ///   The name of the consumer group this event processor is associated with.  Events will be
+        ///   read only in the context of this group.
+        /// </summary>
+        ///
+        public string ConsumerGroup { get; }
+
+        /// <summary>
+        ///   A unique name used to identify this event processor.
+        /// </summary>
+        ///
+        public string Identifier { get; }
+
+        /// <summary>
+        ///   Indicates whether or not this event processor is currently running.
+        /// </summary>
+        ///
+        public bool IsRunning
+        {
+            get
+            {
+                if (_isRunningOverride.HasValue)
+                {
+                    return _isRunningOverride.Value;
+                }
+
+                // TODO: Implement
+                throw new NotImplementedException();
+            }
+
+            protected set => _isRunningOverride = value;
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="EventProcessor{TPartition}"/> class.
+        /// </summary>
+        ///
+        /// <param name="eventBatchSize">The desired number of events to include in a batch to be processed.  This size is the maximum count in a batch; the actual count may be smaller, depending on whether events are available in the Event Hub.</param>
```

Comment:
Possibly?   I don't know that it makes a huge impact on understanding the semantics - I believe developers would still need to read the documentation to clarify since the maximum wait time plays heavily in that.

I don't have a strong option, but since this was the name the board approved, the question comes back to whether or not we feel strongly enough to advocate for the change.   Not sure that I do, but I'm not opposed.  I'll leave that open for thoughts.   @kinelski ?

Reply:
I capitulated here and made the change pending my pitch to the language architect being given a thumbs up.